pkg/cri/resource-manager/policy/builtin/memtier/pools.go

```diff
@@ -245,11 +245,17 @@ func (p *policy) allocatePool(container cache.Container) (Grant, error) {
 
 	request := newRequest(container)
 
+	if p.root.FreeSupply().ReservedCPUs().IsEmpty() && request.CPUType() == cpuReserved {
+		// Fallback to allocating reserved CPUs from shared pool unless reserved
+		// CPU cores are explicitly defined.
+		request.SetCPUType(cpuNormal)
+	}
```

Comment:
The default configuration `ReservedResources: CPU: 750m` leaves `ReservedCPUs()` empty. And so does `CPU > 1000m`, too, because I'm not moving CPUs from the `shared` to the `reserved` set based on the quantity. Therefore this branch is currently part of normal behavior yet it looks like an error.

Do you think we should fill the `reserved` set - not only when Reserved cpuset is defined - but also based on Reserved CPUs quantity?

(I was contemplating this, too... to the extent to playing around with the idea of having a "2nd order" ReservedCPU request and grant, that would contain the CPUs where all reserved containers would be pinned...)

Reply:
> The default configuration `ReservedResources: CPU: 750m` leaves `ReservedCPUs()` empty. And so does `CPU > 1000m`, too, because I'm not moving CPUs from the `shared` to the `reserved` set based on the quantity. Therefore this branch is currently part of normal behavior yet it looks like an error.
> 
> Do you think we should fill the `reserved` set - not only when Reserved cpuset is defined - but also based on Reserved CPUs quantity?

Yes, we should always end up with a reserved cpuset. We should allocate the number of CPUs according to the quantity if that is how CPU reservation is specified.

Reply:
Thanks @klihub! I'll take ceiling of ReservedResources CPU quantity and then use CPU allocator (takeCPUs...) to create cpuset also in that case.